File: surf-api-bukkit/surf-api-bukkit-api/src/main/java/dev/slne/surf/surfapi/bukkit/api/nms/bridges/SurfBukkitNmsStatsBridge.java
package dev.slne.surf.surfapi.bukkit.api.nms.bridges;

import dev.slne.surf.surfapi.bukkit.api.nms.SurfBukkitNmsBridge;
import javax.annotation.ParametersAreNonnullByDefault;
import org.jetbrains.annotations.ApiStatus.NonExtendable;
import org.jetbrains.annotations.NotNull;

@NonExtendable
@ParametersAreNonnullByDefault
public interface SurfBukkitNmsStatsBridge {

  @NotNull
  static SurfBukkitNmsStatsBridge get() {
    return SurfBukkitNmsBridge.get().getStatsBridge();
  }
}

File: surf-api-bukkit/surf-api-bukkit-api/src/main/kotlin/dev/slne/surf/surfapi/bukkit/api/nms/bridges/SurfBukkitNmsStatsBridge.kt
package dev.slne.surf.surfapi.bukkit.api.nms.bridges

import dev.slne.surf.surfapi.bukkit.api.nms.NmsUseWithCaution
import dev.slne.surf.surfapi.core.api.util.requiredService
import org.bukkit.entity.Player

@NmsUseWithCaution
interface SurfBukkitNmsStatsBridge {
    fun getPlayerStatsAsJson(player: Player): String
    fun savePlayerStatsToFile(player: Player)

    companion object {
        val instance = requiredService<SurfBukkitNmsStatsBridge>()
    }
}

@NmsUseWithCaution
val nmsStatsBridge get() = SurfBukkitNmsStatsBridge.instance
